### PR TITLE
chore: PouchDB-find adapters/local refactor to async/await

### PR DESCRIPTION
**Pull request by SourceR85**
Isolated PR from #8860

### CODE REVIEW

File: packages/node_modules/pouchdb-find/src/adapters/local/find/index.js [hunk 1]
```diff
@@ -89,119 +87,113 @@ function find(db, requestDef, explain) {
...
+    return Promise.resolve(queryPlan, opts);
+  }
 
-      if (queryPlan.inMemoryFields.length) {
-        // need to filter some stuff in-memory
-        res.rows = filterInMemoryFields(res.rows, requestDef, queryPlan.inMemoryFields);
-      }
+  const res = indexToUse.name === '_all_docs'
+    ? await doAllDocs(db, opts)
+    : await abstractMapper(db).query.call(db, indexToSignature(indexToUse), opts);
```
Comment (garethbowen) [inline packages/node_modules/pouchdb-find/src/adapters/local/find/index.js:141]: I find this hard to follow. I think it's multiline ternary statements with awaits. A cleaner solution for me is to pull out a named function to do this, eg:

```js
async function queryIndex(db, opts, indexToUse) { // TODO come up with a better fn name
  if (indexToUse.name === '_all_docs') {
    return doAllDocs(db, opts);
  }
  return abstractMapper(db).query.call(db, indexToSignature(indexToUse), opts);
}

...

const res = await queryIndex(db, opts, indexToUse);
```

**Review by garethbowen — CHANGES_REQUESTED** [review]
Nice! Just one suggestion for readability...

File: packages/node_modules/pouchdb-find/src/adapters/local/find/index.js (hunk 1 above)
Comment (SourceR85) [reply packages/node_modules/pouchdb-find/src/adapters/local/find/index.js]: https://github.com/pouchdb/pouchdb/pull/8914/commits/1edb94a6224e82c01f049eb6d708f67e9f127655 should do the trick ;)

**Review by garethbowen — APPROVED** [review]
Thanks!
